src/kakarot/instructions/system_operations.cairo

```diff
@@ -212,19 +212,31 @@ namespace SystemOperations {
         // 0 - size: byte size to copy
         // 1 - offset: byte offset in the memory in bytes
         let (stack, popped) = Stack.pop_n(self=stack, n=2);
-        // TODO: implement loading of the revert reason based on size value,
-        // currently limited by short string size
+
         let size = popped[0];
         let offset = popped[1];
 
         // Load revert reason from offset
-        let (memory, revert_reason_uint256, gas_cost) = Memory.load(memory, offset.low);
-        local revert_reason = revert_reason_uint256.low;
+        let (revert_reason_bytes: felt*) = alloc();
+        let (memory, gas_cost) = Memory.load_n(memory, 32, revert_reason_bytes, offset.low);
```

Comment:
why use `load_n` rather than `load` for 32 bytes? I remember @spapinistarkware mentioned that load is more optimized